Duplicati/UnitTest/UtilityTests.cs

```diff
@@ -204,6 +205,74 @@ public void ParseBool()
                 Assert.IsFalse(Utility.ParseBool(value, returnsFalse));
             }
         }
+
+        [Test]
+        [Category("Utility")]
+        public static void ThrottledStreamRead()
+        {
+            byte[] sourceBuffer = {0x10, 0x20, 0x30, 0x40, 0x50};
+            byte[] destinationBuffer = new byte[sourceBuffer.Length + 1];
+            const int offset = 1;
+            const int bytesToRead = 3;
+
+            using (MemoryStream baseStream = new MemoryStream(sourceBuffer))
+            {
+                const int readSpeed = 1;
+                const int writeSpeed = 1;
+
+                ThrottledStream throttledStream = new ThrottledStream(baseStream, readSpeed, writeSpeed);
+                int bytesRead = throttledStream.Read(destinationBuffer, offset, bytesToRead);
+                Assert.AreEqual(bytesToRead, bytesRead);
+
+                for (int k = 0; k < destinationBuffer.Length; k++)
+                {
+                    if (k < offset)
+                    {
+                        Assert.AreEqual(default(byte), destinationBuffer[k]);
+                    }
+                    else if (k < offset + bytesToRead)
+                    {
+                        Assert.AreEqual(sourceBuffer[k - offset], destinationBuffer[k]);
+                    }
+                    else
+                    {
+                        Assert.AreEqual(default(byte), destinationBuffer[k]);
+                    }
+                }
+            }
+        }
+
+        [Test]
+        [Category("Utility")]
+        public static void ThrottledStreamWrite()

```

Comment:
This test fails without the changes from revision 5d087791d02fe44879decda7ab78dd8d4aa95fa4.